### PR TITLE
[feat]: [PL-24144]: Ng variables get API

### PR DESCRIPTION
**Pull request by nishantsaini20**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32712)

### CODE REVIEW

**Review by piyush-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32712)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: 120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java [hunk 1]
```diff
@@ -42,6 +49,20 @@ public class VariableResource {
   private final VariableService variableService;
   private final VariableMapper variableMapper;
 
+  @GET
+  @Path("{identifier}")
+  public ResponseDTO<VariableResponseDTO> get(@PathParam(IDENTIFIER_KEY) String identifier,
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java:54]: Hide this api from open api spec

Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java:54]: Hide this api from open api spec

File: 120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java [hunk 2]
```diff
@@ -42,6 +49,20 @@ public class VariableResource {
...
+      @QueryParam(ACCOUNT_KEY) String accountIdentifier, @QueryParam(ORG_KEY) String orgIdentifier,
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java:55]: Add NotNull to accountId

Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java:55]: Add NotNull to accountId

File: 120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java [hunk 3]
```diff
@@ -42,6 +49,20 @@ public class VariableResource {
...
+      @QueryParam(PROJECT_KEY) String projectIdentifier) {
+    Optional<VariableResponseDTO> variable =
+        variableService.get(accountIdentifier, orgIdentifier, projectIdentifier, identifier);
+    if (!variable.isPresent()) {
+      throw new NotFoundException(String.format("Variable with identifier [%s] in project [%s] and org [%s] not found",
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java:60]: Should we check and throw this Exception at service level?

Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java:60]: Should we check and throw this Exception at service level?

**Review by nishantsaini20 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32712)* status: 2 file(s) to review. 3. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java` line 54 at r1](https://reviewable.io/reviews/harness/harness-core/32712#-N13EymvtW94xQLTVB6B-r1-54:-N1CaUHIAlrm5teBSjQq:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c8d281a5303c933f7bf78671f8214a1019ae69d8/120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java#L54)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Hide this api from open api spec
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java` line 55 at r1](https://reviewable.io/reviews/harness/harness-core/32712#-N13EymvtW94xQLTVB6B-r1-55:-N1CaUaU4ohfkfK5bR-Y:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c8d281a5303c933f7bf78671f8214a1019ae69d8/120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java#L55)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Add NotNull to accountId
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java` line 60 at r1](https://reviewable.io/reviews/harness/harness-core/32712#-N13EymvtW94xQLTVB6B-r1-60:-N1C_jBB1sGm3IYjJoi_:bn6npgv) ([raw file](https://github.com/harness/harness-core/blob/c8d281a5303c933f7bf78671f8214a1019ae69d8/120-ng-manager/src/main/java/io/harness/ng/core/variable/resources/VariableResource.java#L60)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Should we check and throw this Exception at service level?
</blockquote></details>

Since the response is of type Optional we should leave the decision on the caller on how to handle the empty result.

**Review by meenakshi-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32712#-:-N1Cbzz1A8QNE2YY7Hqv:bnfp4nl)* status: 1 file(s) to review.

**Pull request merged**